lib/test_results_processor.rb

```
@@ -0,0 +1,43 @@
+      pass_rate = (@test_results['pass'].to_f / total_tests * 100).round(2)
+    else
+      pass_rate = 0.0
+    end
+    @test_results['pass_rate'] = pass_rate
+    @test_results
+  end
+
+  # Returns the modified JSON string with the pass rate
+  def to_json
+    @test_results.to_json
+  end
```

rubocop

[warning] `#to_json` requires an optional argument to be parsable via JSON.generate(obj).

(Lint/ToJSON)